Add dynamic archive_size (in GB) to dataset download info message

DerrickUnleashed opened the pull request:
This PR introduces a `self$archive_size` field (in GB) to dataset classes, which is used to dynamically format the `cli_inform()` message shown during dataset initialization. The archive size is emphasized and only shown when greater than 0.1 GB, making messages clearer and standardized across datasets.

Closes #184

DerrickUnleashed commented:
I have also removed the 2016 year from `coco_detection_dataset()` as it is not supported. I have also changed the documentation for `coco_caption_dataset()` stating it supports only 2014 year.

DerrickUnleashed commented:
Consistency improvements implemented :

In `coco_detection_dataset`:
  - Renamed `transforms` to `transform` for consistency.
  - Implemented missing `transform` and `target_transform` parameters.

In `coco_caption_dataset`:
  - Added support for `transform` and `target_transform` parameters.

In Both datasets:
  - Removed system-specific example paths like `root = ~data` for better portability.

cregouby commented:
> @cregouby in `dataset-eurosat` the `$y` returns a torch tensor with value 0, for all these datasets. Is it as expected? Can we change it to return it as integer and add a `target_transform` ? Because `target_transform` is taken as input but never processed.
This is not expected at all, Thanks for raising this !
Sure, please do !

cregouby commented on R/dataset-eurosat.R:
```diff
@@ -35,12 +35,16 @@ eurosat_dataset <- torch::dataset(
   archive_url = "https://huggingface.co/datasets/torchgeo/eurosat/resolve/main/EuroSAT.zip?download=true",
   archive_md5 = "c8fa014336c82ac7804f0398fcb19387",
   split_url = "https://huggingface.co/datasets/torchgeo/eurosat/resolve/main/eurosat-{split}.txt?download=true",
+  archive_size = "<0.1 GB",
```
```suggestion
  archive_size = "90 MB",
```

DerrickUnleashed replied:
Are we going to print 7 MB are do we need to process it and still print <0.1 GB ?  Which is preferred ?

cregouby commented:
> > I don't think so, maybe you are missing the Write access. I've created[ the issue 190 for that](https://github.com/mlverse/torchvision/issues/190).
> 
> ```r
> > devtools::check_win_devel()
> Building windows version of torchvision (0.6.0.9000)
> ℹ Using R-devel with win-builder.r-project.org.
> Email results to daniel@rstudio.com?
> 
> 1: No way
> 2: Uhhhh... Maybe?
> 3: Yup
> 
> Selection: 
> Enter an item from the menu, or 0 to exit
> Selection: 1
> 
> > revdepcheck::cloud_check()
> ℹ Building package tarball
> Error: Cloud error (403): Forbidden
> Message: Forbidden
> 
> > revdepcheck::revdep_check()
> ── INIT ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── Computing revdeps ──
> ── INSTALL ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 2 versions ──
> Installing CRAN version of torchvision
> Installing DEV version of torchvision
> Installing 1 packages: torch
> ── CHECK ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 7 packages ──
> ✔ bundle 0.1.2                           ── E: 1     | W: 0     | N: 0                                                                                                                         
> ✔ cito 1.1                               ── E: 0     | W: 0     | N: 0                                                                                                                         
> ✔ deepregression 2.2.0                   ── E: 0     | W: 0     | N: 0                                                                                                                         
> ✔ geodl 0.2.0                            ── E: 0     | W: 0     | N: 1                                                                                                                         
> ✔ luz 0.4.0                              ── E: 1     | W: 0     | N: 0                                                                                                                         
> ✔ mlr3torch 0.2.1                        ── E: 1     | W: 0     | N: 0                                                                                                                         
> ✔ torchdatasets 0.3.1                    ── E: 0     | W: 0     | N: 0                                                                                                                         
> OK: 7                                                                                                                                                                                        
> 
> BROKEN: 0
> Total time: 14 min
> ── REPORT ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
> Writing summary to 'revdep/README.md'
> Writing problems to 'revdep/problems.md'
> Writing failures to 'revdep/failures.md'
> Writing CRAN report to 'revdep/cran.md'
> > 
> ```
> 
> Is this fine @cregouby ? Can I push this? (625 changes)

`revdepcheck` is only for posit employees. forget about it.

DerrickUnleashed commented:
Are we going to print 90 MB are do we need to process it and still print <0.1 GB ? Which is preferred ?

DerrickUnleashed commented on R/dataset-fer.R:
```diff
@@ -38,19 +39,20 @@ fer_dataset <- dataset(
 
+    cli_inform("{.cls {class(self)[[1]]}} Dataset (~{.emph {self$archive_size}}) will be downloaded and processed if not already available.")
+
     cli_inform("{.cls {class(self)[[1]]}} Dataset will be downloaded and processed if not already available.")
```
Oops Sorry about that.

DerrickUnleashed commented on R/dataset-eurosat.R:
```diff
@@ -176,7 +183,8 @@ eurosat100_dataset <- torch::dataset(
   archive_md5 = "c21c649ba747e86eda813407ef17d596",
-  split_url = "https://huggingface.co/datasets/torchgeo/eurosat/resolve/main/eurosat-100-{split}.txt?download=true"
+  split_url = "https://huggingface.co/datasets/torchgeo/eurosat/resolve/main/eurosat-100-{split}.txt?download=true",
+  archive_size = "<0.1 GB"
```
Sure I'll update as suggested.

DerrickUnleashed commented:
> **praise** Nice move ! Much more explicit messages ! **todo** small changes **improvement** We currently inform of ` cli_inform("{.cls {class(self)[[1]]}} Dataset will be downloaded and processed if not already available.")` before the test of download, i.e. even if the end-user doesn't want to download, with an explicit download = FALSE, which is misleading. Could we move the cli_inform() within the `if (download)` ? **improvement** turn all `archive_size = "<0.1 GB"` to an explicit value. else the cli message will be " ~ < 0.1GB " which is non sense. **missing**

Thanks for the suggestions!

I agree I have moved the `cli_inform` to within the `if(download)` section to make it more meaningful. 

Sure, I have updated such that it prints out specific sizes so that it makes sense instead of <0.1 GB

cregouby commented on tests/testthat/test-dataset-eurosat.R:
```diff
@@ -84,17 +86,18 @@ test_that("eurosat100_dataset derivatives download and prepare correctly", {
   skip_on_cran()
   skip_if_not_installed("torch")
 
+  skip_if(Sys.getenv("TORCHVISION_ALLOW_LARGE_TESTS") != "1",
+        "Skipping test: set TORCHVISION_ALLOW_LARGE_TESTS=1 to enable tests requiring large downloads.")
```
**todo** please use the more robust test for the environment variable. see https://github.com/mlverse/torch/blob/eb5f9031aee3292ec64e37efd549c72d5c22cb78/tests/testthat.R#L4 (along all tests)
**suggestion** I would rename variable into `TEST_LARGE_DATASETS` so that we have a room later on for `TEST_LARGE_MODELS`  (along all tests)

DerrickUnleashed replied:
Thanks for the suggestion! Sure I'll update the environment variable name to `TEST_LARGE_DATASETS`

cregouby commented on tests/testthat/test-dataset-mnist.R:
```diff
@@ -139,10 +141,9 @@ test_that("tests for the emnist dataset", {
 
   expect_error(
-      ds <- qmnist_dataset(dir, split = "nist"),
+      ds <- qmnist_dataset(tempfile()),
```
**todo** please keep the correct split so that we know error is not due to missing split

cregouby reviewed (changes requested):
**todo**  few changes on tests

DerrickUnleashed commented:
> > @cregouby Since now I'm using an environment variable for tests with size > 300 MB. Is there a way to manually set this environment variable in the GitHub workflow, as the tests are also skipped here?
> 
> Yes we could. My idea was to set the environment variable on developer laptop, to perform CI -CD test much faster. What's your proposed strategy ?

That makes sense, setting the variable locally definitely speeds up development. My suggestion was to ensure that large dataset tests are still covered in CI, perhaps by enabling TEST_LARGE_DATASETS=1 in a nightly workflow (similar to regular R-CMD checks). This would involve updating [this workflow file](https://github.com/mlverse/torchvision/blob/main/.github/workflows/R-CMD-check.yaml), which would help us catch issues across all operating systems. I’ve seen cases where something passes on one OS but fails on another, so this could add a useful safety net.
